admin-app/image.go

```diff
@@ -87,10 +114,15 @@ func postImageHandler(app_settings common.AppSettings, database database.Databas
 		err = database.AddImage(uuid.String(), file.Filename, alt_text)
 		if err != nil {
 			log.Error().Msgf("could not add image metadata to db: %v", err)
-			err := os.Remove(image_path)
-			if err != nil {
+			os_err := os.Remove(image_path)
```

Comment:
I feel like this should be deffered straight after the call to `c.SaveUploadedFile(...)` - that way if anything else goes wrong in between, the file is also not kept

Reply:
The problem here is, if you defer it it will be executed whether you succeeded with the upload or not. We want to remove the file if, and only if, the creation of the meta data entry failed.

Reply:
I guess we could check the error in the deferred function, but that might be too unreadable. Fine with this